apps/system/lockscreen/js/widgets/clock/lockscreen_clock_widget.js

```diff
@@ -50,6 +52,56 @@
     this.logger.debug('Clock updated', now);
   };
 
+  LockScreenClockWidget.prototype.updateAlarm =
+	function() {
+    this.fetchAlarmData()
+    .then((alarmData) => {
+      // if there is no alarm
+      if(!alarmData) {
+        this.resources.elements.alarm.classList.add('no-alarms');
+        this.resources.elements.alarm.classList.remove('has-alarm');
+        return;
+      }
+
+      var type = 'AM',
+          hour = parseInt(alarmData.hour, 10);
+
+      // Decide wheter to use AM or PM
+      if(hour > 12) {
+        type = 'PM';
+        hour = hour - 12;
+      }
+
+      var fullAlarmTime = hour + ':' + alarmData.minute + type;
+      this.resources.elements.alarmtime.textContent = fullAlarmTime;
+      this.resources.elements.alarm.classList.remove('no-alarms');
+      this.resources.elements.alarm.classList.add('has-alarm');
+      this.logger.debug('Alarm updated to :', fullAlarmTime);
+    });
+	};
+
+  /*
+   * Gets the Alarm Data from the DataStore.
+   */
+  LockScreenClockWidget.prototype.fetchAlarmData = function() {
+    return navigator.getDataStores('alarms')
+      .then((stores) => {
+        return [stores[0].getLength(), stores[0]];
+      }).then(([len, store]) => {
+        if(0 === len) {
+          return;
+        }
+        return store.get(1);
+      }).catch(function(e){
+        console.error('Error occured while getting alarm data : ', e);
```

Comment:
You better re-throw the error after log it, since in this way the following promises concates it could be interrupted by the error, too. And, if you're addressing general errors in the whole chain, you don't need to catch it here, but put the catch at the end of the chain. That is, rather than:

    return getDataStores().then().then().catch().then();

You can write it as:

    return  return getDataStores().then().then().then().catch();

Since to put a `catch` in the middle of the chain will make you miss any error after that. So if you're not intentionally to do that, put it at the end is a convention.